src/components/Highlight/index.tsx

```diff
@@ -25,18 +29,41 @@ const Highlighted = styled.span`
  * Highlight component is meant to be a simple component to display text with highlighted search.
  * Component will render a SPAN or series of SPAN with the content and highlights
  */
-export const Highlight: React.FC<HighlightProps> = ({ text, highlight, ignoreCase = true, theme }) => {
+export const Highlight: React.FC<HighlightProps> = ({ text, matches, highlight, ignoreCase = true, theme }) => {
+  if (!matches && !highlight) {
+    throw new Error('You must set either indices or highlight');
+  }
+
+  if (matches) {
+    let lastMatch = 0;
```

Comment:
Could you also validate `matches` to make sure it contains only numbers and the second one is more than first?

Some examples of odd behaviors:

![image](https://user-images.githubusercontent.com/11414342/70915137-fd114b00-2021-11ea-80d7-9f726f8d13ab.png)

![image](https://user-images.githubusercontent.com/11414342/70915307-4bbee500-2022-11ea-8681-d7e0e3c230fa.png)

![image](https://user-images.githubusercontent.com/11414342/70915682-ef0ffa00-2022-11ea-8ed8-1a325d720fe7.png)

![image](https://user-images.githubusercontent.com/11414342/70915978-8bd29780-2023-11ea-88e3-78315e8d5c91.png)

![image](https://user-images.githubusercontent.com/11414342/70916033-a0169480-2023-11ea-9cf7-7a1b2892f0d8.png)


```ts
const matches = "string"; // or {}, Date, function, etc.
<Highlight text="Hello World! Searching for World!" matches={matches} />

// TypeError: matches.map is not a function
```

PS:
It would be great to cover those cases by unit tests...